services/src/main/java/org/keycloak/authentication/authenticators/x509/ValidateX509CertificateUsername.java

```diff
@@ -63,7 +65,7 @@ public void authenticate(AuthenticationFlowContext context) {
         if (config == null) {
             logger.warn("[ValidateX509CertificateUsername:authenticate] x509 Client Certificate Authentication configuration is not available.");
             context.getEvent().error(Errors.USER_NOT_FOUND);
-            Response challengeResponse = errorResponse(Response.Status.UNAUTHORIZED.getStatusCode(), "invalid_request", "Configuration is missing.");
+            Response challengeResponse = ErrorPage.error(context.getSession(), context.getAuthenticationSession(), Response.Status.UNAUTHORIZED, Messages.MISSING_CONFIGURATION);
```

Comment:
Missing configuration should return a 500 internal server error. Message returned to user should be generic, it should not expose internal error messages to end-users.